File: python/graphqlRequest.py
import urllib.request
import urllib.parse
import time
import hashlib
import hmac

CONTENTTYPE_JSON = "application/json"
GRAPHAPI_DATALIMIT = 1024*1024*1024
HTTP_CONNECTIONTIMEOUT = 1000
HTTP_READWRITETIMEOUT = 3000
METHOD_POST = "POST"
CHARACTER_ENCODING = "utf-8"

key = "[API-Key-Here]"
secret = "[Secret-Key-Here]"

url = "https://clearcaster.c2.wowza.com/graphql"
domain = "clearcaster.c2.wowza.com"

requestStr = "{\"query\":\"query allEncoders { allEncoders { id name } }\"}"
data = requestStr.encode('ascii')

currTime = int(round(time.time() * 1000))

currTimeBytes = bytes(str(currTime), encoding=CHARACTER_ENCODING)
secretBytes = bytes(secret, encoding=CHARACTER_ENCODING)
domainBytes = bytes(domain, encoding=CHARACTER_ENCODING)

sig1 = hmac.new(secretBytes, currTimeBytes, digestmod=hashlib.sha256).digest()
sig2 = hmac.new(sig1, domainBytes, digestmod=hashlib.sha256).digest()

authorizationHeader = "HMAC-SHA256, Credential=" + key + ", SignedHeaders=host;x-date, Signature=" + sig2.hex()

headers = {
	'Authorization': authorizationHeader,
	'X-Date': str(currTime),
	'Content-Type': CONTENTTYPE_JSON,
	'Accept': CONTENTTYPE_JSON
	}

req = urllib.request.Request(url, data, headers)
with urllib.request.urlopen(req) as response:
   print(response.read().decode(CHARACTER_ENCODING))

File: php/graphqlRequest.php
<html>
<head>
<title>Wowza ClearCaster GraphQL PHP Test</title>
</head>
<body>
<?php

$key = "[API-Key-Here]";
$secret = "[Secret-Key-Here]";

$url = "https://clearcaster.c2.wowza.com/graphql";
$domain = "clearcaster.c2.wowza.com";

$queries = <<<'QUERIES'
query allEncoders {
    allEncoders {
        id
        name
    }
}
QUERIES;

function makeRequest($url, $domain, $accessKey, $secretKey, $queries, $operationName = null, $variables = null)
{
    $body = "{";

    if (isset($queries))
    {
        $queriesFixed = str_replace("\n", " ", $queries);
        $queriesFixed = str_replace("\t", " ", $queriesFixed);
        $body .= "\"query\":\"" . $queriesFixed . "\",";
    }
    
    if (isset($operationName))
        $body .= "\"operationName\":\"" . $operationName . "\",";
    
    if (isset($variables))
        $body .= "\"variables\":" . json_encode($variables) . ",";

    $body = substr($body, 0, -1);

    $body .= "}";

    $ch = curl_init($url);

    curl_setopt($ch, CURLOPT_POST, 1);
    curl_setopt($ch, CURLOPT_POSTFIELDS, $body);
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);

    $milliseconds = (string)round(microtime(true) * 1000);

    $tmp = hash_hmac('sha256', $milliseconds, $secretKey);
    $signature = hash_hmac('sha256', $domain, hex2bin($tmp));
    $authHeader = 'HMAC-SHA256, Credential=' . $accessKey . ',  SignedHeaders=host;x-date, Signature=' . $signature;
    
    curl_setopt($ch, CURLOPT_HTTPHEADER, array(
        'Content-Type: application/json',
        'Authorization: ' . $authHeader,
        'X-Date: ' . $milliseconds
    ));
    
    $response = curl_exec($ch);
    
    return $response;
}

$variables = array();

$response = makeRequest($url, $domain, $key, $secret, $queries, "allEncoders", $variables);

echo "RESPONSE[allEncoders]: " . $response . "<br/><br/>";

?>
</body>
</html>